### ISSUE TITLE
tooth pattern

### ISSUE THREAD

**Issue by megbalk** [obophenotype/uberon#2761]
I revised tooth pattern to better fit with what @wdduncan and I have discussed. I'll make a PR and would love feedback.

**Comment by megbalk** [comment]
how do I create an annotation on "has_exact_synonym" to have "synonym_type_property" and "abbreviation"?

**Comment by megbalk** [comment]
@rays22 how do I test this?

**Comment by wdduncan** [comment]
@megbalk Have you run `sh run.sh make IMP=false test` locally? Does it complete w/o error?

Also, you can run `sh run.sh make ../patterns/definitions.owl -B IMP=false` to generate your `definitions.owl` file. See the **DOSDP generate: Turning the template tables into OWL axioms** section on this page: https://oboacademy.github.io/obook/tutorial/dosdp-odk/

**Comment by megbalk** [comment]
@wdduncan I was talking with a mammalogist, and this is how they write out teeth:
M3 = third secondary upper molar
dp2 = second deciduous lower premolar

Would it make sense to have a dentition pattern first that then an upper and lower pattern builds off of, or does it not matter (i.e., can use either independently)?

**Comment by wdduncan** [comment]
@megbalk I'm not sure if it is better to have dentition or position. The FMA used dentition first (e.g. [upper first secondary molar tooth](https://ontobee.org/ontology/FMA?iri=http://purl.obolibrary.org/obo/FMA_55811). But, I don't know why.

**Comment by markengelstad** [comment]
There are many ways to do this, but if you want to be able to use a scalable extensible system that can model anything, then you should include , in order:

Dentition first (if organism has > 1 set of teeth)  :  because it’s consistent and logical, I prefer “Primary” and “Secondary”  rather than “deciduous” and “permanent”.

Followed by arch:  Upper and Lower.  (I think this makes more sense than maxillary and mandibular for reasons I Cld explain. )

Then morphology/  form :  Incisor , canine, premolar, molar and the sequence from the midline. : 1,2,3 or first, second, third



Mark Engelstad
________________________________

**Comment by wdduncan** [comment]
correction on my comment above   
> The FMA used dentition first

The FMA seems to be put the **arc** first (e.g., [upper first secondary molar tooth](https://ontobee.org/ontology/FMA?iri=http://purl.obolibrary.org/obo/FMA_55811)) ... sorry about that.

For animals that regenerate teeth over their lifetime (e.g., sharks) does the primary/secondary distinction make sense to use?

**Comment by markengelstad** [comment]
Yeah, agreed. for those species that do not have distinctly different sets of teeth/dentition, you wouldn’t need to specify which set.

Mark Engelstad
________________________________

**Comment by megbalk** [comment]
ok, so agreed that it should be: dentition, arch, tooth number and type. So, I'll keep the pattern as is: one of upper and lower (arch) and one for dentition based on the outputs from the arch.

**Comment by wdduncan** [comment]
> dentition, arch, tooth number and type

E.g.: `upper first secondary molar tooth` -> `secondary upper first molar tooth`

**Comment by megbalk** [comment]
I will test my patterns then!

**Comment by megbalk** [comment]
I'm getting the following errors but don't see what is wrong. Could I get some help?

<img width="567" alt="Screenshot 2023-01-18 at 11 41 01 PM" src="https://user-images.githubusercontent.com/6617332/213373400-65bd9e92-15ed-4b51-b1af-0e0ecbd7612b.png">

**Comment by anitacaron** [comment]
What is your complete YAML file?

**Comment by megbalk** [comment]
@anitacaron what do you mean?

**Comment by anitacaron** [comment]
@megbalk I need to check the complete pattern configuration file to be able to help. Is there a PR you're working on?

**Comment by megbalk** [comment]
@anitacaron I'm working on branch issue-2761

**Comment by wdduncan** [comment]
@megbalk Do you want to keep this open?

**Comment by megbalk** [comment]
sure, I won't be able to get to it for a bit fyi

**Comment by ANiknejad** [comment]
Hello,

this publication https://pubmed.ncbi.nlm.nih.gov/27872325/ talks about **tooth family unit**  in polyphyodonts, that includes a functional tooth (ft), a replacement tooth (RT) and a dental lamina (dl).

> To study dl transcriptome profiles at different stages, we performed RNA-seq on the total extracted RNA. After alignment to the alligator genome ([Green et al. 2014](https://pmc.ncbi.nlm.nih.gov/articles/PMC5119682/#bibr7-0022034516667724)), we calculated the reads per kilobase per million reads mapped (RPKM) levels. Hierarchical clustering showed that dl samples collected from the same stage were grouped tightest ([Fig. 1C](https://pmc.ncbi.nlm.nih.gov/articles/PMC5119682/#fig1-0022034516667724)), suggesting that our dissection method was reliable. However, **different stages also showed co-clustering: cap-stage and bell-stage samples were grouped together, and bud-stage and RT-bell samples grouped together.** These significant expression profile differences with respect to the point of tooth initiation (cap-stage vs. bud-stage) imply that different gene pathways must be activated to start this process. Principal component analysis also showed that samples collected from the same stage wereclustered together ([Fig. 1D](https://pmc.ncbi.nlm.nih.gov/articles/PMC5119682/#fig1-0022034516667724)).


https://trace.ncbi.nlm.nih.gov/Traces/sra?study=SRP080802
SRP080802 reports libraries for both 'dental lamina' and for 'replacement tooth', and 'dental lamina' are all reported such as 'Pre-initiation stage' in SRA metadata, while library names are as detailed as:

- Dental lamina Early Growth stage  ---> bud?
- Dental lamina Initiation stage ----> cap?
- Dental lamina Pre-initiation stage ---> bell?

Should for example 'dental lamina bud-stage' be a new class in Uberon?

@markengelstad do you consider here that having a dedicated class for 'replacement tooth' may be useful to make distinction between functional tooth and replacement tooth?
So far, and as far as I can see, there is only UBERON:0007115 deciduous tooth available in UBERON to reflect 'replacement tooth', and its definition is restricted to mammals, means, diphyodonts

Sorry in advance for this long and detailed message, but it is also an opportunity to have this issue still active!

**Comment by wdduncan** [comment]
The Oral Health and Disease Ontology has classes for various types of [prosthetic teeth](https://ontobee.org/ontology/OHD?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FOHD_0000302).

**Comment by ANiknejad** [comment]
Sorry in advance but I am lost here, not sure the 2 last messages are related to my post, there are none prosthetic tooth in the 
complex tooth family unit of polyphyodonts, but functional tooth (ft), a replacement tooth (RT) and a dental lamina (dl), each one involved in natural tooth development stages

OHD:0000206 functional tooth
Definition: A natural, modified or prosthetic tooth

This class does not allow to distinct the ft and the rt in polyphyodonts

'tooth replacement' sounds common in literature, see https://pubmed.ncbi.nlm.nih.gov/23788284/
_Biology of tooth replacement in amniotes_  (2013)

**Comment by wdduncan** [comment]
Sorry, I thought 'replacement tooth' referred to things like tooth implants. Sorry for the misunderstanding.

In the OHD, we distinguish between primary and secondary teeth, but, of course, this is human specific. You may need to distinguish between deciduous and non-deciduous teeth. 

Perhaps there is a better label for 'replacement tooth', but I can't think of one.

**Comment by wdduncan** [comment]
@aleixpuigb do you know was going to work on adding tooth surfaces?
